internal/store/event.go

```diff
@@ -0,0 +1,77 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store/model"
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type Event interface {
+	InitialMigration() error
+
+	Create(ctx context.Context, orgId uuid.UUID, event *api.Event) error
+	List(ctx context.Context, orgId uuid.UUID, listParams ListParams) (*api.EventList, error)
+	DeleteOlderThan(ctx context.Context, cutoffTime time.Time) (int64, error)
+}
+
+type EventStore struct {
+	db           *gorm.DB
+	log          logrus.FieldLogger
+	genericStore *GenericStore[*model.Event, model.Event, api.Event, api.EventList]
+}
+
+// Make sure we conform to Event interface
+var _ Event = (*EventStore)(nil)
+
+func NewEvent(db *gorm.DB, log logrus.FieldLogger) Event {
+	genericStore := NewGenericStore[*model.Event, model.Event, api.Event, api.EventList](
+		db,
+		log,
+		model.NewEventFromApiResource,
+		(*model.Event).ToApiResource,
+		model.EventsToApiResource,
+	)
+	return &EventStore{db: db, log: log, genericStore: genericStore}
+}
+
+func (s *EventStore) InitialMigration() error {
+	if err := s.db.AutoMigrate(&model.Event{}); err != nil {
+		return err
+	}
+
+	// Create a composite index on involved_object_kind and involved_object_name
+	if err := s.db.Exec("CREATE INDEX IF NOT EXISTS idx_event_involved_obj_kind_name ON events (involved_object_kind, involved_object_name)").Error; err != nil {
```

Comment:
BTW, you can create composite index by model declaration in gorm: https://gorm.io/docs/indexes.html#Composite-Indexes

Reply:
Good call!  Fixed

Reply:
@coderabbitai couldnt you identify this issue?